backend/src/common/middleware/jwt.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Injectable,
+  NestMiddleware,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { Request, Response, NextFunction } from 'express';
+
+@Injectable()
+export class JwtMiddleware implements NestMiddleware {
```

Comment:
You don't need this file

Reply:
However, you do need to make sure you validate the token when it comes and to be able to mark your graphql queries as private or public


For that you'd need to use Nestjs Guards: https://docs.nestjs.com/guards
This is a good example how to do it with rest: https://docs.nestjs.com/security/authentication#enable-authentication-globally
With graphql it'll be a bit different